sdk/cosmos/azure-spring-data-cosmos-core/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -189,6 +193,14 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
         return toDomainObject(domainType, response.getItem());
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> void generateIdIfNullAndAutoGenerationEnabled(T originalItem, Class<?> type) {
+        CosmosEntityInformation<?, ?> entityInfo = CosmosEntityInformation.getInstance(type);
+        if (entityInfo.shouldGenerateId() && ReflectionUtils.getField(entityInfo.getIdField(), originalItem) == null) {
+            ReflectionUtils.setField(entityInfo.getIdField(), originalItem, UUID.randomUUID().toString());
```

Comment:
I remember we had an issue with UUID generation in activityId causing log pauses due to lack of real randomness entropy. 
Do other spring plugin use auto generated UUID? I wonder if we should use fast uuid generation for this (similar to activityId in the SDK)

Reply:
@moderakh - can you please point to the code where we auto generate activity id in the SDK code, may be we can implement something similar in spring if it helps with the performance.

Reply:
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Utils.java#L438-L440

Reply:
Thanks for the code @moderakh , however the above TimeBasedUUIDGenerator is being brought in by uuid external dependency which is currently in the banned dependency list. 
I also checked our V2 async code, and we use the randomUUID generator to generate UUIDs. 

In my opinion, it is not worth going through internal shading process for this. What are your thoughts?

Reply:
One suggestion, write a for loop with UUID.randomUUID() and measure the time for each UUID generation, if the time looks good then go with UUID.randomUUID()

```java
int start = System.currentMillis();
for (int i = 0; i < 1_000_000; i++) {
  UUID.randomUUID();
}
int end = System.currentMillis();
System.out.println(end - start);
```

preferably run this on a linux VM where you don't have mouse movements, etc, as those generate random entropy which is used by /dev/random.